linked_list.c

```diff
@@ -1,105 +1,197 @@
 #include <assert.h>
 #include "linked_list.h"
 
+
 void ll_init(struct ll_node *node) {
     assert(node != NULL);
     // TODO: Implement your solution here.
+
+    node->next = NULL;
+    node->prev = NULL;
 }
 
 bool ll_has_next(struct ll_node *node) {
     assert(node != NULL);
     // TODO: Implement your solution here.
-    return false;
+
+    return node->next != NULL;
 }
 
 bool ll_has_prev(struct ll_node *node) {
     assert(node != NULL);
     // TODO: Implement your solution here.
-    return false;
+
+    return node->prev != NULL;
 }
 
 struct ll_node *ll_next(struct ll_node *node) {
     assert(node != NULL);
     // TODO: Implement your solution here.
+
+    if(ll_has_next(node)) {
+        return node->next;
+    }
     return NULL;
 }
 
 struct ll_node *ll_prev(struct ll_node *node) {
     assert(node != NULL);
     // TODO: Implement your solution here.
+
+    if(ll_has_prev(node)) {
+        return node->prev;
+    }
     return NULL;
 }
 
 size_t ll_size(struct ll_node *head) {
     assert(head != NULL);
     // TODO: Implement your solution here.
-    return 0;
+
+    size_t size = 1;
+    
+    while(ll_has_next(head)) {
+        size++;
+        head = ll_next(head);
+    }
+    return size;
 }
 
 struct ll_node *ll_head(struct ll_node *list) {
     assert(list != NULL);
     // TODO: Implement your solution here.
-    return NULL;
+
+    while(ll_has_prev(list)) {
+        list = ll_prev(list);
+    }
+    return list;
 }
 
 struct ll_node *ll_tail(struct ll_node *list) {
     assert(list != NULL);
     // TODO: Implement your solution here.
-    return NULL;
+
+    while(ll_has_next(list)) {
+        list = ll_next(list);
+    }
+    return list;
 }
 
 struct ll_node *ll_get(struct ll_node *node, size_t index) {
     assert(node != NULL);
     // TODO: Implement your solution here.
-    return NULL;
+
+    // is this assertion needed?
+    // assert(index < size);
+
+    for(long unsigned int i = 0; i < index; i++) {
+        node = ll_next(node);
+        if(node == NULL) { //how do I compare 2 linked list nodes? seg fault
```

Comment:
You don't need to compare the nodes.  Simply comparing the pointers (as you've done here with a `NULL` check) is sufficient, and is roughly equivalent to using `==` [instead of `Object#equals()`] in Java.